src/App.tsx

```diff
@@ -1,18 +1,228 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useState,
+} from 'react';
+
+import classNames from 'classnames';
+
+import {
+  CustomTodo,
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Filter } from './types/Filter';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import Form from './components/Form/Form';
+import TodoList from './components/TodoList/TodoList';
+import Warning from './components/Warning/Warning';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todosFromTheServer, setTodosFromTheServer] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorCount, setErrorCount] = useState(0);
+  const [todosToShow, setTodosToShow] = useState(todosFromTheServer);
```

Comment:
you don't need a separate state for it
you can just save it in const:
`const todosToShow = filterTodos()`
no need to pass a filter as argument - it is available because of `closure`
it will be recalculated every time `filter` or `todosFromTheServer` (any state actually - wrap it in useMemo if you don't want to recalculate it when it's not necessary) changed